src/fields/hooks/afterRead/relationshipPopulationPromise.ts

```diff
@@ -53,6 +53,16 @@ const populate = async ({
     if (!relationshipValue) {
       // ids are visible regardless of access controls
       relationshipValue = id;
+    } else if (field.select) {
+      const newRelationShipValue = {
+        id: relationshipValue.id,
+      };
+      field.select.forEach((fieldName) => {
```

Comment:
Do we want to support nesting fields?

`select: ['group.someField`]` for example. We could split the fieldName on the "." and build up the objects accordingly.

Reply:
I also thought about adding a way to select nested fields, but we would have to decide how to select arrays.
Either we would have to select the index of the array, or we would have to select the field name of the value in the array and then select the whole array.

Example of the first option:

```js
const relationshipField = {
  name: 'purchase',
  type: 'relationship',
  relationTo: ['products', 'services'],
  select: ['name', 'price', 'images.0.url'],
};
// returns
{
  id: '123',
  name: 'Product Name',
  price: 100,
  images: [
    {
      id: '456',
      url: 'https://example.com/image.jpg'
    }
  ]
}
```

Example of the second option:

```js
const relationshipField = {
  name: 'purchase',
  type: 'relationship',
  relationTo: ['products', 'services'],
  select: ['name', 'price', 'images.url'],
};

// returns
{
  id: '123',
  name: 'Product Name',
  price: 100,
  images: [
    {
      id: '456',
      url: 'https://example.com/image.jpg'
    },
    {
      id: '789',
      url: 'https://example.com/image2.jpg'
    }
  ]
}
```